src/contracts/storage/tickmap.rs

```diff
@@ -249,11 +261,25 @@ impl Tickmap {
             "tick initialize tick again"
         );
 
+        if returned_chunk == 0 {
+            self.store_initialized_chunk(pool_key, chunk)
+        }
+
         self.bitmap.insert(
             (chunk, pool_key),
             &flip_bit_at_position(returned_chunk, bit),
         );
     }
+
+    pub fn store_initialized_chunk(&mut self, pool_key: PoolKey, chunk: u16) {
+        let mut vec = self.initialized_chunks.get(pool_key).unwrap_or_default();
+        vec.push(chunk);
```

Comment:
Will these chunks not be saved multiple times here if I initialize multiple ticks in one chunk? (If this is true, then checking if the chunk is added may be too expensive)